### PR TITLE
Implement new accounts API query options

### PR DESCRIPTION
**Pull request by butonic**
We updated the api to follow the google recommendations and use LDAP as the storage backend.

- we introduce a new ldap attribute `ownclouduuid` to store a uuid for ocis accounts

part of https://github.com/owncloud/ocis-glauth/issues/13

### CODE REVIEW

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/store/filesystem/store.go  7
- pkg/service/v0/service.go  16
         

Clones added
============
- pkg/store/filesystem/store.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5488614&bid=17986943)

File: pkg/account/accounts.go [hunk 1]
```diff
@@ -17,9 +17,16 @@ var (
 type RegisterFunc func(*config.Config) Manager
 
 // Manager is an accounts service interface
+// TODO email and username might not be unique?
 type Manager interface {
-	// Read a record
-	Read(key string) (*proto.Record, error)
+	// Read a record by uuid
+	Read(uuid string) (*proto.Record, error)
```
[the remark this answers is not in the document]
Comment (butonic) [reply pkg/account/accounts.go]: now using string constants ...

**Comment by butonic** [comment]
@refs any Idea how we could provision users? 
I guess we need a cli to add them?
This is a marshaled account I wrote by hand:
```
{
  "key": "1-2-3-4",
  "payload": {
    "account": {
      "issuer": "https://localhost:9200",
      "password": "relativity",
      "standard_claims": {
        "sub": "1234",
        "preferred_username": "einstein",
        "name": "Albert Einstein",
        "given_name": "Albert",
        "family_name": "Einstein"
      }
    }
  }
}
```

password is not encrypted ... hm need to fix that

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/store/filesystem/store.go  12
- pkg/service/v0/service.go  18
         

Clones added
============
- pkg/store/filesystem/store.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5488614&bid=17986943)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/store/filesystem/store.go  12
- pkg/service/v0/service.go  18
         

Clones added
============
- pkg/store/filesystem/store.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5488614&bid=17986943)

**Comment by refs** [comment]
> @refs any Idea how we could provision users?
> I guess we need a cli to add them?
> This is a marshaled account I wrote by hand:
> 
> ```
> {
>   "key": "1-2-3-4",
>   "payload": {
>     "account": {
>       "issuer": "https://localhost:9200",
>       "password": "relativity",
>       "standard_claims": {
>         "sub": "1234",
>         "preferred_username": "einstein",
>         "name": "Albert Einstein",
>         "given_name": "Albert",
>         "family_name": "Einstein"
>       }
>     }
>   }
> }
> ```
> 
> password is not encrypted ... hm need to fix that

I think the import command was meant to do that 🤔

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/service.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5488614&bid=17986943)

**Comment by refs** [comment]
Adding the Blocked label because #22 will be merged first. Once we merge all PR on ocis, ocis-accounts, ocis-settings and ocis-proxy, work on this will resume.

**Comment by IljaN** [comment]
> I think the import command was meant to do that

This is strictly for migrating users from another source. The accounts-service should have a seperate command to create users from cli.

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/provider/ldap.go  14
- pkg/provider/ldap_test.go  4
- pkg/service/v0/service.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5488614&bid=17986943)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/provider/ldap_test.go  4
- pkg/provider/ldap.go  14
- pkg/service/v0/service.go  6
         

Clones added
============
- pkg/service/v0/service.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5488614&bid=17986943)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/provider/ldap_test.go  4
- pkg/provider/ldap.go  14
- pkg/service/v0/service.go  6
         

Clones added
============
- pkg/service/v0/service.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5488614&bid=17986943)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/provider/ldap.go  14
- pkg/service/v0/service.go  6
- pkg/provider/ldap_test.go  4
         

Clones added
============
- pkg/service/v0/service.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5488614&bid=17986943)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/provider/ldap_test.go  4
- pkg/provider/ldap.go  14
- pkg/service/v0/service.go  6
         

Clones added
============
- pkg/service/v0/service.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5488614&bid=17986943)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/provider/bleve.go  24
- pkg/provider/tokenizer.go  1
- pkg/service/v0/service.go  4
         

Clones added
============
- pkg/provider/bleve.go  2
- pkg/service/v0/service.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5488614&bid=17986943)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/service.go  12
- pkg/provider/ldap_test.go  4
- pkg/provider/ldap.go  16
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5488614&bid=17986943)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/service.go  12
- pkg/provider/ldap.go  16
- pkg/provider/ldap_test.go  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5488614&bid=17986943)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/provider/ldap.go  16
- pkg/service/v0/service.go  13
- pkg/provider/ldap_test.go  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5488614&bid=17986943)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/provider/ldap.go  16
- pkg/service/v0/service.go  13
- pkg/provider/ldap_test.go  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5488614&bid=17986943)

**Comment by butonic** [comment]
It turns out upstream does not implement LdapAdd or LdapDelete messages, so we will change the architecture to a sync based approach where all users and groups are stored by ocis-accounts directly.

We will revisit this when time permits. For now we are going with a sync based approach in a new PR.

**Pull request closed without merging**